src/schemas/category-schema.js

```diff
@@ -20,4 +22,11 @@ var CategorySchema = new Schema({
 
 });
 
+CategorySchema.pre('save', function(next) {
```

Comment:
I think a pre `create` would be better, so we don't have to use this middleware each time we modified the category.